Dockerfile

```diff
@@ -1,33 +1,60 @@
-FROM ubuntu:14.04
-MAINTAINER Luc Belliveau <luc.belliveau@nrc-cnrc.gc.ca>
-
-# Install dependencies
-ARG DEBIAN_FRONTEND=noninteractive
-RUN apt-get update
-RUN apt-get install -y git apache2 php5 libapache2-mod-php5 php5-mysql php5-gd php5-curl curl
-RUN curl -sS https://getcomposer.org/installer | php -- --install-dir=/usr/local/bin --filename=composer
-
-# Enable Apache mod_rewrite
-RUN a2enmod rewrite
-
-# Modify Apache config for Elgg
-RUN echo '<Directory /var/www/html>\nDirectoryIndex index.php\nOptions FollowSymLinks MultiViews\nAllowOverride All\nOrder allow,deny\nallow from all\n</Directory>\n' | sed '/^<VirtualHost \*:80>/r /dev/stdin' /etc/apache2/sites-available/000-default.conf > /etc/apache2/sites-available/tmp
-RUN mv /etc/apache2/sites-available/tmp /etc/apache2/sites-available/000-default.conf
-
-# Modify Apache config to output access and error log to stdio
-# So we can see the output using `docker-compose logs` or directly with `docker-compose up`)
-RUN sed -i '/ErrorLog ${APACHE_LOG_DIR}\/error.log/c\ErrorLog \/dev\/stderr' /etc/apache2/apache2.conf
-RUN sed -i '/ErrorLog ${APACHE_LOG_DIR}\/error.log/c\ErrorLog \/dev\/stderr' /etc/apache2/sites-available/000-default.conf
-
-# Uncomment to also see access log (rebuild with docker-compose up --build)
-# RUN sed -i '/CustomLog ${APACHE_LOG_DIR}\/access.log combined/c\CustomLog \/dev\/stdout combined' /etc/apache2/sites-available/000-default.conf
-
-COPY . /var/www/html
-WORKDIR /var/www/html
-
+# First stage, install composer and its dependencies and fetch vendor files
+FROM alpine:latest
+RUN mkdir /app && apk --no-cache add \
+  php5 \
+  php5-dom \
+  php5-phar \
+  php5-gd \
+  php5-json \
+  php5-mysql \
+  php5-openssl \
+  php5-xml \
+  php5-zlib \
+  curl
+RUN curl -sS https://getcomposer.org/installer | php5 -- --install-dir=/usr/local/bin --filename=composer
+RUN ln -s /usr/bin/php5 /usr/bin/php
+WORKDIR /app
+COPY composer.json composer.json /app/
 ARG COMPOSER_ALLOW_SUPERUSER=1
 ARG COMPOSER_NO_INTERACTION=1
 RUN composer install
 
+# Second stage, build usable container
+FROM alpine:latest
+LABEL maintainer="Luc Belliveau <luc.belliveau@nrc-cnrc.gc.ca>"
+RUN \
+  apk --no-cache add \
+    apache2 \
+    apache2-utils \
+    php5-apache2 \
+    php5-curl \
+    php5-dom \
+    php5-gd \
+    php5-json \
+    php5-mysql \
+    php5-xml \
+  && mkdir -p /var/www/html/vendor \
+  && mkdir -p /var/www/html/data \
```

Comment:
I've been meaning to change it in the Dockerfile so might as well do it now, the data dir should really be outside of a directory served by apache. just /data would work.